Convert Repository to Python Package.

anissingh opened the pull request:
## What?
Converted this repository into a Python package that can be installed with `pip` through GitHub (note that this package is **not** currently available on PyPI). Focused more on the technical side of the package conversion (i.e., getting the package up and running) rather than design decisions, although some design decisions were made.

## Why?
Converting this repository into a Python package is an easy way to integrate it into the MarkUs scanned exam pipeline.

## How?
Restructured the files in the repository to adhere to the structure of a Python package. Followed [PEP 518](https://peps.python.org/pep-0518/) and used a `pyproject.toml` file to specify the build system requirements and metadata for the package. In particular, `requirements.txt` was deleted and the runtime dependencies contained within it were moved to the `pyproject.toml` file.

## Future Improvements
The package should be refactored and design decisions should be made that lead to easy to understand and sustainable code.

anissingh commented:
Note that one of my commit messages is incorrect. For commit [af07641](https://github.com/MarkUsProject/markus-exam-matcher/pull/2/commits/af07641e41b7f503543e023969bc28d3b8f9b7c6), the issue has been closed, yet the version of`urllib3` still needs to be downgraded.

david-yz-liu reviewed (commented):
@anissingh this is all good, but I'm not sure whether the functions are being properly exported from the module. Using subfolders is good, but to make them submodules (where you can import from the individual files) you may need a `__init__.py` in each one, though I'm a bit fuzzy on the details.

Also, you should still include sample usage at this point (even as the API is changing) to ensure the base functionality. It seems like you've gotten rid of the main block in `read_chars.py`. This is fine, but it's important to provide (and document) an alternative. It's okay if this takes the form of Python shell interactions, e.g.

```python
>>> from markus_exam_matcher.image_processing.read_chars import my_function
>>> my_function('../examples/my_example.jpg')
123456789
```

anissingh commented:
@david-yz-liu The functions are being exported correctly from the modules, as with Python 3.3+, `__init__.py` files are no longer explicitly required for directories to be recognized as packages. However, I did some more reading on this, and it appears that leaving out the `__init__.py` file creates something called a namespace package instead of a regular package, which is not what I was trying to do for the subpackages. Also, it violates the second guiding design principle of Python [(The Zen of Python)](https://peps.python.org/pep-0020/), which states its preference for being explicit over implicit. Consequently, I've incorporated a change to add `__init__.py` files to the subdirectories.

As for your second comment, I've added a `run` method that allows users of the package to simply call this method and not worry about what is going on behind the scenes, and updated the `README` file to include a sample usage. This is probably best for quick integration into MarkUs. Originally, I had a slightly different vision in mind for the package, so the package should definitely be refactored in the future.
